bitwarden_license/src/Sso/Controllers/AccountController.cs

```diff
@@ -407,106 +402,96 @@ public async Task<IActionResult> Logout(string logoutId)
             }
 
             OrganizationUser orgUser = null;
-            if (orgId.HasValue)
+            var organization = await _organizationRepository.GetByIdAsync(orgId);
+            if (organization == null)
             {
-                var organization = await _organizationRepository.GetByIdAsync(orgId.Value);
-                if (organization == null)
+                throw new Exception(_i18nService.T("CouldNotFindOrganization", orgId));
+            }
+            
+            // Try to find OrgUser via existing User Id (accepted/confirmed user) or Email + OrgId (invited user)
+            var orgUsersByOrgId = await _organizationUserRepository.GetManyByOrganizationAsync(orgId, null);
```

Comment:
I think you may want to split this up into 2 separate DB calls; the list of org users could be in the 1k's, really you need to get a single by User Id, if currentUser is not null, then if that result is null OR current user is null, fetch by Email (separate call).

The logic I see here looks like it will apply an OR, what if there is an OrgUser with the UserID *and* a different OrgUser with that email address? The logic could not distinguish them nor give preference to the UserId match first.

Reply:
Updated! Split into two separate db calls and prioritized `existingUser.Id` search result.